coda.io.website.json

```diff
@@ -0,0 +1,31 @@
+{  
+    "providerId":"coda.io",
+    "providerName":"Coda",
+    "serviceId":"website",
+    "serviceName":"Coda Site",
+    "version": 1,
+    "syncPubKeyDomain" : "coda.io",
+    "logoUrl":"https://cdn.coda.io/icons/png/color/coda-192.png",
+    "description":"Allows serving Coda documents from your own domain",
+    "variableDescription":"`host` refers to the subdomain that the user wishes to use (not just limited to 'www'). `subdomain` refers to the subdomain for Coda's custom domain server. `docId` refers to the coda document used for custom domain. `verification` refers to the TXT host for doc ownership verification, differs based on A or CNAME. `ipAddress` refers to the proper IP for an apex domain to point to.",
+    "records":[  
+       {  
+          "type":"CNAME",
+          "host":"%host%",
```

Comment:
Please do not use `%host%` as variable as it is a reserved name and typically you don't need to use it in the template.
See [spec Section 6.7](https://github.com/Domain-Connect/spec/blob/master/Domain%20Connect%20Spec%20Draft.adoc#hostname-in-template) for details how it works.

I am also not sure how this template should work with both APEX A-Record and a CNAME on subdomain.
Maybe you are looking for 2 different templates for a subdomain and APEX or you need groups to selectively apply only part of the records?

Reply:
Ah thanks, sounds like we probably do want either multiple templates or groupId. Maybe the simple option is two group names like `cname` and `apex`

Just to clarify the intended behavior:

For a subdomain like `www.example.com`:
- Set a `CNAME` record on `www.example.com` pointing to `<variable>.coda.io`. The CNAME `pointsTo` value is variable so that we can use different CNAME in testing environments vs production.
- Set a `TXT` record at `_coda.www.example.com` with a verification blob to prove to Coda the the document owner really controls the subdomain
- _DO NOT_ set an `A` record. Rely entirely on CNAME

For an apex domain like `example.com`:
- Set three `A` records on `example.com` with a few fixed IPs for redundancy
- Set a `TXT` record at `_coda.example.com` with a verification blob to prove to Coda the the document owner really controls the domain
- _DO NOT_ set a `CNAME` (which should be impossible anyway?...)

Reply:
I think you'd be better off with 2 templates, like in those templates https://github.com/Domain-Connect/Templates/blob/master/zenfolio.customsubdomain.json and https://github.com/Domain-Connect/Templates/blob/master/zenfolio.customdomain.json.
1 template with groups won't be working for you, as CNAME is not possible on APEX and `hostRequired` would need to be set, on the other side it blocks usage of this template for APEX.